### DISCUSSION TITLE
sergen code generation problem

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6738] [Help]
Hi,

on Visual Studio 2022 I have:

1. created a new project as explained in this tutorial: 
     https://serenity.is/docs/tutorials/movies/movies); 
     serenity 6.6.3 was installed
2. added migration for Tenant table as explained in this tutorial:
    https://serenity.is/docs/tutorials/multi_tenancy/adding_tenants_table_and_tenantid_field
3. generated code with sergen as explained in this tutorial:
    https://serenity.is/docs/tutorials/multi_tenancy/generating_code_for_tenants_table

the problem is that when a run the project and, after login, select menu item Administration/Tenant nothing happens; 
no datagrid with tenants is displayed; 
no errors on javascript console;
only a grey page!

I've done the same steps next week, on another test project, and all worked fine.

Any idea?

![Schermata 2023-03-27 alle 20 15 27](https://user-images.githubusercontent.com/14870650/228030788-22c69e72-341e-484e-b814-6d46d201efaa.jpg)

**Comment by elenaaralla** [comment]
Hi,

more info: on test project that works, was installed serenity 6.5.1

**Comment by VictorTomaili** [comment]
You have need to update sergen with the command `dotnet tool update sergen`
Check the version of the sergen by `dotnet sergen`
It should be same or upper version than your project. 

Check if you have not installed the global tool sergen. When you run `sergen` it runs global tool. If you use `dotnet sergen` project scoped tool works. Be ensuring about you need to use `dotnet sergen`. 

Check your sergen.json file has the line `"Extends": "defaults@6.6.0"` for the latest version.

**Comment by elenaaralla** [comment]
Hi,

> Please continue in discussion, this seems not a bug. 
Ok, sorry!

>Your TenantPage.ts file has some differences. Delete it and re-create it again.

I've deleted _**TenantPage.ts**_ file and re-generated it; this is the content of the file after regeneration:

```
import { initFullHeightGridPage } from '@serenity-is/corelib/q';
import { TenantGrid } from './TenantGrid';

export default function pageInit() {
    initFullHeightGridPage(new TenantGrid($('#GridDiv')).element);
}
```
If I change:

```
export default function pageInit() {
    initFullHeightGridPage(new TenantGrid($('#GridDiv')).element);
}
```
with
```
$(function () {
    initFullHeightGridPage(new TenantGrid($('#GridDiv')).element);
});
```
It works; 

Now, the question is:  how come sergen generates the code that doesn't work?

Thank you,
Elena

**Reply by VictorTomaili** [reply]
Probably the version or some package mismatch. Seems it is stable in the latest version.

**Reply by elenaaralla** [reply]
Indeed it seems I have sergen and nuget packages 6.6.3
What kind of package mismatch are you referring to? Any idea where I can look for?
this is a new solutions, not an upgraded one!

thanks

**Reply by VictorTomaili** [reply]
I don't have any ideas, I was just telling theoretically.

**Reply by elenaaralla** [reply]
ok, not the answer I was lookin for, but thanks!
Elena.

**Reply by martiza** [reply]
I have the same thing 
everything is updated and to stated version 

this worked for  me by @elenaaralla 


`
$(function () {
    initFullHeightGridPage(new TenantGrid($('#GridDiv')).element);
});
`

**Reply by ahmadfauzanarif** [reply]
this worked for me to 
thank @elenaaralla 
![image](https://user-images.githubusercontent.com/71299117/229007811-8ebdbd99-48a4-4b7a-8cb6-3e14fa26541b.png)

**Comment by volkanceylan** [comment]
Sergen was generating the new recommended code for Page.ts with export default for 6.6+, but we forgot to delete Views/Shared/_GridPage.cshtml and Modules/Common/GridPage/*.cs files from Serene so your generated page was using old versions of those files instead of the new ones in Serenity.Extensions, and the pageInit function was never getting called. Sorry for the trouble. Either delete those files, or install Serene 6.6.3.1+ VSIX
